google/oauth2/aio/credentials.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""OAuth 2.0 Asynchronous Credentials.
+
+This module provides credentials based on OAuth 2.0 access and refresh tokens.
+These credentials usually access resources on behalf of a user (resource
+owner).
+
+Specifically, this is intended to use access tokens acquired using the
+`Authorization Code grant`_ and can refresh those tokens using a
+optional `refresh token`_.
+
+Obtaining the initial access and refresh token is outside of the scope of this
+module. Consult `rfc6749 section 4.1`_ for complete details on the
+Authorization Code grant flow.
+
+.. _Authorization Code grant: https://tools.ietf.org/html/rfc6749#section-1.3.1
+.. _refresh token: https://tools.ietf.org/html/rfc6749#section-6
+.. _rfc6749 section 4.1: https://tools.ietf.org/html/rfc6749#section-4.1
+"""
+
+from google.auth.aio import credentials
+
+
+class Credentials(credentials.Credentials):
+    """Asynchronous Credentials using OAuth 2.0 access and refresh tokens.
+
+    """
+
+    def __init__(self, token):
```

Comment:
I don't think the token should be a part of the constructor.

Reply:
Why not? It's a part of the synchronous version of this method here: https://github.com/googleapis/google-auth-library-python/blob/main/google/oauth2/credentials.py#L73

Reply:
I don't love that API as `token` is dependent on `expiry`. It may not be obvious that both are required, and when expiry is omitted the token is always `valid`.

My 2c, leave it for now and maybe we introduce it using a builder pattern.

Reply:
I've introduced a builder class here: https://github.com/googleapis/google-auth-library-python/pull/1559/commits/1bb65171b2caedd39498bd8cb2fddbda4d66b7f7. Let me know what you think!


- IIUC from your comment, both `token `and `expiry` are required. Given that the class expects the token to never expire / user's responsibility to re-initialize the credentials with a new token and make the request (if needed), do we want to introduce `expiry` and `setExpiry` at this point?

    -  if we are to move forward with this approach, we should be emitting this information to the user and also lmk that the credentials' class will not attempt to automatically refresh the token before making the request.

Reply:
I think it should be in the same style as the existing code: https://github.com/googleapis/google-auth-library-python/blob/main/google/oauth2/credentials.py#L282.

Reply:
I think a static token class makes sense for your use case.

It is similar to https://github.com/googleapis/google-auth-library-python/blob/main/google/oauth2/credentials.py#L578 but does not support refresh.

Reply:
Addressed!